### PR TITLE
fixes #22174; fixes destructor examples

### LINKED ISSUE #22174: `lent` example does not compile
### Description

https://nim-lang.org/docs/destructors.html#lent-type

```nim
proc `[]`*(x: Tree; i: int): lent Tree =
  result = x.kids[i]
  # borrows from 'x', this is transformed into:
  result = addr x.kids[i]
  # This means 'lent' is like 'var T' a hidden pointer.
  # Unlike 'var' this hidden pointer cannot be used to mutate the object.
```

although this perhaps was meant to demonstrate the mechanism with the extra `addr`, the code does not compile making for a bad copy-paste experience when trying to use `lent`

### Nim Version

web site

### Current Output

_No response_

### Expected Output

_No response_

### Possible Solution

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ringabout**
fixes #22174

### CODE REVIEW

File: doc/destructors.md [hunk 1]
```diff
@@ -731,7 +731,7 @@ used to specialize the object traversal in order to avoid deep recursions:
       if x.left != nil: s.add(x.left)
       if x.right != nil: s.add(x.right)
       # free the memory explicit:
-      dispose(x)
+      `=dispose`(x)
```
Comment (Araq) [inline doc/destructors.md:734]: The =dispose hook is not officially supported. It should be `=destroy(x[])` or something like that.

File: doc/destructors.md [hunk 2]
```diff
@@ -731,7 +731,7 @@ used to specialize the object traversal in order to avoid deep recursions:
       if x.left != nil: s.add(x.left)
       if x.right != nil: s.add(x.right)
       # free the memory explicit:
```
Comment (Araq) [inline doc/destructors.md]:
```suggestion
      # free the memory explicitly:
```

File: doc/destructors.md (hunk 1 above)
Comment (ringabout) [reply doc/destructors.md:734]:
```nim
template `=dispose`*[T](x: owned(ref T) =
  ...
  let hdrSize = align(sizeof(RefHeader), alignment)
  alignedDealloc(p -! hdrSize, alignment)
```
Isn't `=dispose(x)` used to free the pointer `x`. Is it equivalent to `=destroy(x[])`?

Comment (Araq) [reply doc/destructors.md:734]: Ok.

**Pull request merged**